### ISSUE TITLE
Generating the Sequence File from Multiple Consensus Genomes

### ISSUE THREAD

**Issue by FatihSarigol** [gphocs-dev/G-PhoCS#51]
Hi!
This is not really an issue about the program, but instead of asking via email, I thought perhaps having it here might help someone else, too.
I have a few whole genomes (which are in same coordinates with the variants applied to the reference, which consists of over 1000 contigs) from different individuals, and I want to run G-PhoCS on them. Could you suggest an easy way to generate a proper format input sequence file from them? I don't have much time to write a code to do that right now, but if nobody else has a similar code, I would be happy to share it here once I write it myself one day.
Any help is much appreciated!
Thanks

**Comment by FatihSarigol** [comment]
Thank you for your message!
GLACtools has a script to export ACF files (which contains allele counts for either a single individual or a group of individuals (population)) as G-PhoCS format. 
I haven't tried it, yet, but it also can convert single sample VCF to ACF, so could be a way even though a long one seemingly.
Any alternative ideas to go from fasta files with variants already applied?
Thanks!

**Comment by gphocs-dev** [comment]
You'll probably have to write up a custom script for that. I typically end up using different custom scripts for different data sets.

**Comment by grenaud** [comment]
@FatihSarigol  I am the author of glactools. I just saw this, converting VCF to ACF should be straightforward. The only problem is getting an outgroup or ancestral if need be. There is a perl script to convert contiguous chunks to gphocs output. Let me know if you run into any issues.

**Comment by FatihSarigol** [comment]
Hello @grenaud 
Thank you for your comment! I recently wrote my own script to merge and convert fasta files of different samples into the format that G-PhoCS requires.

If anyone else needs to take that road, too, I'd be happy to share my code; I'll eventually put it on my github, but can still make it better for other users.

**Comment by grenaud** [comment]
Yes glactools is not really designed for fasta files. It targets mostly genotyping or single bases from BAM files.

**Comment by gphocs-dev** [comment]
Thanks for both of your inputs.
--Ilan

**Comment by mvsantosherp** [comment]
> Hello [@grenaud](https://github.com/grenaud) Thank you for your comment! I recently wrote my own script to merge and convert fasta files of different samples into the format that G-PhoCS requires.
> 
> If anyone else needs to take that road, too, I'd be happy to share my code; I'll eventually put it on my github, but can still make it better for other users.

Hello @FatihSarigol 

I've been struggling to get a Gphocs input, and also using VCFs using whole genome sequencing. It would be great if you shared your script. Thanks in advance

**Comment by FatihSarigol** [comment]
> > Hello [@grenaud](https://github.com/grenaud) Thank you for your comment! I recently wrote my own script to merge and convert fasta files of different samples into the format that G-PhoCS requires.
> > If anyone else needs to take that road, too, I'd be happy to share my code; I'll eventually put it on my github, but can still make it better for other users.
> 
> Hello [@FatihSarigol](https://github.com/FatihSarigol)
> 
> I've been struggling to get a Gphocs input, and also using VCFs using whole genome sequencing. It would be great if you shared your script. Thanks in advance

Hi @mvsantosherp 
It was a very manual code for a very specific case,
Variant versions of the reference genome fasta files with SNPs changed (using bcftools consensus), for multiple samples of the same species.
It goes something like this:
```

#Split each fasta file which should have all and every bases of all chromosomes in one single line
#This line does that, changing a fasta file into that single line format:
for i in *fa; do sed '/^>/ d' ${i} | tr -d '\n' > ${i%.fa}.2split.fa;done
#This line renames them
for i in *2split.fa;do mv ${i} ${i%.snpIUPAC.2split.fa}.fa;done

#Then for each sample with fasta file, I split the single line whole genome sequence into sooo many pieces of files, 2 examples are below:
split --bytes=47250 --additional-suffix=ANT --numeric-suffixes=1 --suffix-length=5 ANTARCTIC.fa
#With this line I add the sample name into the split files which were only sequence pieces before
sed -i 's/^/ANTARCTIC /' *ANT

#Another example, same thing will be done for all samples, here you can decide how long the sequences will be, I had chosen arbitrarily 47250, only to use the first 1000 of each later:
split --bytes=47250 --additional-suffix=BER --numeric-suffixes=1 --suffix-length=5 BERN.fa
sed -i 's/^/BERN /' *BER

#Then I add new lines to the ends of the split files which are needed in GPhocs format as separation
sed -i -e '$a\' *x*

#Generate Gphocs format headers and split that file, this was again for the format requirement
paste <(paste <(for i in `seq 50221`; do echo locus; done) <(seq 50221) | sed -e 's/\t//g') <(paste <(for i in `seq 50221`; do echo 11; done) <(for i in `seq 50221`; do echo 1000; done)) | sed -e 's/\t/ /g' > NUMS
#Split the file to be also used while combining each sample sequence piece
split -l 1 --additional-suffix=ABC --numeric-suffixes=1 --suffix-length=5 NUMS

#Add new line to the beginning of each header split line
sed -i '1i\\' *ABC

#Concatenate all of the split pieces which all start with x and they will be concatenated in the name/number order
cat x* > TOTAL

#Add main header which is the number of sequences
sed -i '1 i\50221' TOTAL

#Get first 1K of each line which is something I wanted to do using the first 1000 bases in each split sequence
cut -c1-1000 TOTAL > GPHOCSready

#Note that splitting a genome into pieces like this for many samples will create thousands of files, so it is better to do it in a new folder with no other files needed, and don't do ls unless necessary, or do ls | head  and ls | tail. Hope it is useful/helpful
```
